blockchain/blockchain.go

```diff
@@ -1721,9 +1723,9 @@ func (bc *BlockChain) insertChain(chain types.Blocks) (int, []interface{}, []*ty
 				winner[j], winner[len(winner)-1-j] = winner[len(winner)-1-j], winner[j]
 			}
 			// Import all the pruned blocks to make the state available
-			bc.chainmu.Unlock()
+			bc.mu.Unlock()
```

Comment:
I don't know it is a good approach. What about applying the same approach as WriteBlockWithState adn writeBlockWithState? 
Calling locking and unlocking in InsertChain?